sources/packages/api/src/services/institution/institution.service.spec.ts

```diff
@@ -1,13 +1,62 @@
 require("../../../env_setup");
 import { Test, TestingModule } from "@nestjs/testing";
+import * as faker from "faker";
 import { InstitutionService } from "./institution.service";
 import { DatabaseModule } from "../../database/database.module";
 import { BCeIDService } from "../bceid/bceid.service";
 import { ConfigService } from "../config/config.service";
 import { UserService } from "../user/user.service";
+import { Institution, User } from "../../database/entities";
+import { DatabaseService } from "../../database/database.service";
+
+const factory = async (
+  userService: UserService,
+  service: InstitutionService,
+): Promise<[Institution, User]> => {
+  const user = userService.create();
+  user.email = faker.internet.email();
+  user.firstName = faker.name.firstName();
+  user.lastName = faker.name.lastName();
+  user.userName = `${faker.random.uuid()}@bceid`;
+  const institution = service.create();
+  institution.legalOperatingName = faker.company.companyName();
+  institution.operatingName = institution.legalOperatingName;
+  institution.guid = faker.random.uuid();
+  institution.establishedDate = faker.date.past();
+  institution.website = faker.internet.url();
+  institution.institutionAddress = {
+    phone: faker.phone.phoneNumber(),
+    addressLine1: faker.address.streetAddress(),
+    city: "Victoria",
+    country: "Canada",
+    postalCode: "V8V1M9",
+    provinceState: "BC",
+  };
+  institution.primaryEmail = faker.internet.email();
+  institution.primaryPhone = faker.phone.phoneNumber();
+  institution.institutionPrimaryContact = {
+    primaryContactEmail: faker.internet.email(),
+    primaryContactLastName: faker.name.lastName(),
+    primaryContactFirstName: faker.name.firstName(),
+    primaryContactPhone: faker.phone.phoneNumber(),
+  };
+  institution.legalAuthorityContact = {
+    legalAuthorityEmail: faker.internet.email(),
+    legalAuthorityLastName: faker.name.lastName(),
+    legalAuthorityFirstName: faker.name.firstName(),
+    legalAuthorityPhone: faker.phone.phoneNumber(),
+  };
+  institution.regulatingBody = "ICBC";
+  institution.users = [user];
+  await service.save(institution);
+
+  return [institution, user];
+};
 
 describe("InstitutionService", () => {
```

Comment:
This service is now interacting with the database, are we going to keep those "Unit Tests" executing always to it should be good to skip them?

Reply:
I will include this as integration test later.

Reply:
Test skipped